modin/pandas/__init__.py

```diff
@@ -105,21 +105,41 @@
 def _update_engine(publisher: Parameter):
     global dask_client
     from modin.config import StorageFormat, CpuCount
+    from modin.config.envvars import IsExperimental
+    from modin.config.pubsub import ValueSource
 
-    if publisher.get() == "Ray":
+    if (
+        publisher.get_value_source() == ValueSource.DEFAULT
+        and StorageFormat.get() == "Omnisci"
+    ):
+        publisher.put("Native")
+        IsExperimental.put(True)
+    elif publisher.get() == "Ray":
```

Comment:
I suggest taking the filling part fully above the following `if engine == A do_A() else do_B()` logic.

So it would be something like
```python
if StorageFormat.get() == "Omnisci" and Engine.get_value_source() == ValueSource.DEFAULT:
    Engine.put("Native")
    IsExperimental.put(True)
elif Engine.get() == "Native" and StorageFormat.get_value_source() == ValueSource.DEFAULT:
    StorageFormat.put("Omnisci")
    IsExperimental.put(True)
```
and then go on with checks and initializations as before